kubectl-fdb/cmd/buggify_no_schedule_test.go

```diff
@@ -0,0 +1,214 @@
+						Expect(len(tc.ExpectedInstancesInNoSchedule)).To(BeNumerically("==", len(resCluster.Spec.Buggify.NoSchedule)))
+					},
+					Entry("Adding the same instance.",
+						testCase{
+							Instances:                     []string{"instance-1"},
+							ExpectedInstancesInNoSchedule: []string{"instance-1"},
+						}),
+					Entry("Adding different instance.",
+						testCase{
+							Instances:                     []string{"instance-2"},
+							ExpectedInstancesInNoSchedule: []string{"instance-1", "instance-2"},
+						}),
+					Entry("Adding multiple instances.",
+						testCase{
+							Instances:                     []string{"instance-2", "instance-3"},
+							ExpectedInstancesInNoSchedule: []string{"instance-1", "instance-2", "instance-3"},
+						}),
+				)
+			})
+		})
+
+		When("removing process group from no-schedule list from a cluster", func() {
+			var kubeClient client.Client
+
+			BeforeEach(func() {
+				cluster.Spec.Buggify.NoSchedule = []string{"instance-1", "instance-2", "instance-3"}
+				scheme := runtime.NewScheme()
+				_ = clientgoscheme.AddToScheme(scheme)
+				_ = fdbv1beta2.AddToScheme(scheme)
+				kubeClient = fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(cluster, &corev1.PodList{}).Build()
+			})
+
+			type testCase struct {
+				Instances                     []string
+				ExpectedInstancesInNoSchedule []string
+			}
+
+			DescribeTable("should remove all targeted processes from the no-schedule list",
+				func(tc testCase) {
+					err := updateNoScheduleList(kubeClient, clusterName, tc.Instances, namespace, false, true, false)
+					Expect(err).NotTo(HaveOccurred())
+
+					var resCluster fdbv1beta2.FoundationDBCluster
+					err = kubeClient.Get(ctx.Background(), client.ObjectKey{
+						Namespace: namespace,
+						Name:      clusterName,
+					}, &resCluster)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(tc.ExpectedInstancesInNoSchedule).To(Equal(resCluster.Spec.Buggify.NoSchedule))
+					Expect(len(tc.ExpectedInstancesInNoSchedule)).To(BeNumerically("==", len(resCluster.Spec.Buggify.NoSchedule)))
+				},
+				Entry("Removing single instance.",
+					testCase{
+						Instances:                     []string{"instance-1"},
+						ExpectedInstancesInNoSchedule: []string{"instance-2", "instance-3"},
+					}),
+				Entry("Removing multiple instances.",
+					testCase{
+						Instances:                     []string{"instance-2", "instance-3"},
+						ExpectedInstancesInNoSchedule: []string{"instance-1"},
+					}),
+			)
+
+		})
+
+		When("clearing no-schedule list", func() {
+			var kubeClient client.Client
+
+			BeforeEach(func() {
+				cluster.Spec.Buggify.NoSchedule = []string{"instance-1", "instance-2", "instance-3"}
+				scheme := runtime.NewScheme()
+				_ = clientgoscheme.AddToScheme(scheme)
+				_ = fdbv1beta2.AddToScheme(scheme)
+				kubeClient = fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(cluster, &corev1.PodList{}).Build()
```

Comment:
Why are we adding here an empty Pod list if the `updateNoScheduleList` method doesn't query any Pods?